stl/inc/regex

```diff
@@ -4078,8 +4014,33 @@ bool _Matcher3<_BidIt, _Elem, _RxTraits, _It, _Alloc>::_Match_pat(_Node_base* _N
                             }
                         }
                     } else {
-                        _Failed = !_Do_rep(_Node, _Greedy, 0);
-                        _Next   = nullptr;
+                        if (_Node->_Min > 0 || (_Greedy && !_Longest && _Node->_Max != 0)) { // try a rep first
+                            // set up stack unwinding for greedy matching or loop val restoration
+                            auto _Frame_idx = _Push_frame(
+                                _Node->_Min == 0 ? _Rx_unwind_ops::_Loop_greedy : _Rx_unwind_ops::_Loop_restore_vals,

```

Comment:
Suggestion: Extracting the code would make this easier to read (even though it wouldn't save a line).